test/transaction_test.dart

```diff
@@ -125,62 +123,28 @@ void main() {
     });
 
     test('May intentionally rollback transaction', () async {
-      var reached = false;
-      await conn.transaction((c) async {
-        await c.query('INSERT INTO t (id) VALUES (1)');
-        c.cancelTransaction();
```

Comment:
All the tests previously using `cancelTransaction()` look a bit weird now since the only way to explicitly do a rollback in v3 is to throw an exception. In some cases, that causes compile-time errors if the test is unchanged since the transaction then has a type of `Never`. I've used the test API to expect the exception instead.